apps/interop-rxjs-signal/src/app/list/photos.component.ts

```diff
@@ -75,39 +72,17 @@ import { PhotoStore } from './photos.store';
           <div>No Photos found. Type a search word.</div>
         </ng-template>
         <footer class="text-red-500">
-          {{ vm.error }}
+          {{ store.error() }}
         </footer>
       </section>
     </ng-container>
   `,
-  providers: [provideComponentStore(PhotoStore)],
   host: {
     class: 'p-5 block',
   },
 })
-export default class PhotosComponent implements OnInit {
-  store = inject(PhotoStore);
-  readonly vm$ = this.store.vm$.pipe(
-    tap(({ search }) => {
-      if (!this.formInit) {
-        this.search.setValue(search);
-        this.formInit = true;
-      }
-    })
-  );
-
-  private formInit = false;
-  search = new FormControl();
-
-  ngOnInit(): void {
-    this.store.search(
-      this.search.valueChanges.pipe(
-        skipWhile(() => !this.formInit),
-        debounceTime(300),
-        distinctUntilChanged()
-      )
-    );
-  }
+export default class PhotosComponent {
+  store = injectAutoSignal(PhotoStore);
```

Comment:
very nice, you are creating a viewmodel but without the ngIf 

And you do it this way to create a lazy subscription to signal ? Instead of using the store signal directly ? right ?

Reply:
Yes